tests/components/somfy_mylink/test_config_flow.py

```diff
@@ -315,71 +328,103 @@ async def test_options_no_entities(hass):
 
-    config_entry = MockConfigEntry(
+    mock_imported_config_entry = MockConfigEntry(
         domain=DOMAIN,
-        data={CONF_HOST: "1.1.1.1", CONF_PORT: 12, CONF_SYSTEM_ID: 46},
+        data={
+            CONF_HOST: "1.1.1.1",
+            CONF_PORT: 1234,
+            CONF_SYSTEM_ID: 456,
+            CONF_DEFAULT_REVERSE: True,
+            CONF_ENTITY_CONFIG: {"cover.xyz": {CONF_REVERSE: False}},
+        },
     )
-    config_entry.add_to_hass(hass)
+    mock_imported_config_entry.add_to_hass(hass)
+
+    mock_status_info = {
+        "result": [
+            {"targetID": "1.1", "name": "xyz"},
+            {"targetID": "1.2", "name": "zulu"},
+        ]
+    }
 
     with patch(
         "homeassistant.components.somfy_mylink.SomfyMyLinkSynergy.status_info",
-        return_value={
-            "result": [
-                {
-                    "targetID": "a",
-                    "name": "Master Window",
-                    "type": 0,
-                }
-            ]
-        },
+        return_value=mock_status_info,
     ):
-        assert await hass.config_entries.async_setup(config_entry.entry_id)
+        assert await hass.config_entries.async_setup(
+            mock_imported_config_entry.entry_id
+        )
         await hass.async_block_till_done()
-        result = await hass.config_entries.options.async_init(config_entry.entry_id)
+
+        assert mock_imported_config_entry.options == {
+            "reversed_target_ids": {"1.2": True}
+        }
+
+        result = await hass.config_entries.options.async_init(
+            mock_imported_config_entry.entry_id
+        )
         await hass.async_block_till_done()
         assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
         assert result["step_id"] == "init"
 
         result2 = await hass.config_entries.options.async_configure(
             result["flow_id"],
-            user_input={"default_reverse": True, "entity_id": "cover.master_window"},
+            user_input={"target_id": "1.2"},
         )
 
         assert result2["type"] == data_entry_flow.RESULT_TYPE_FORM
         result3 = await hass.config_entries.options.async_configure(
             result2["flow_id"],
-            user_input={"reverse": False},
+            user_input={"reverse": reversed},
         )
 
         assert result3["type"] == data_entry_flow.RESULT_TYPE_FORM
 
         result4 = await hass.config_entries.options.async_configure(
             result3["flow_id"],
-            user_input={"default_reverse": True, "entity_id": None},
+            user_input={"target_id": None},
         )
         assert result4["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
 
-        assert config_entry.options == {
-            "default_reverse": True,
-            "entity_config": {"cover.master_window": {"reverse": False}},
-            "entity_config_version": 1,
+        # Will not be altered if nothing changes
+        options_copy = mock_imported_config_entry.options.copy()
```

Comment:
Is there a shallow copy in the core somewhere that we're working around? I don't understand this part.

Reply:
The code looks right in `async_update_entry`.  I'll dig some more.

```
        if options is not UNDEFINED and entry.options != options:  # type: ignore
            changed = True
            entry.options = MappingProxyType(options)
```

Reply:
When we create the `MappingProxyType` it's just doing a shallow copy, if I remember correctly. Maybe that's the issue?

Reply:
Thanks that helped me find the issue.  It is a shallow copy in the config flow. 🤦 

All fixed